File: y-py/tests/test_helper.py
import y_py as Y

def transact(self,callback):
    with self.begin_transaction() as txn:
        return callback(txn)



def exchange_updates(docs):
    for d1 in docs:
        for d2 in docs:
            if d1 != d2:
                state_vector = Y.encode_state_vector(d1)
                diff = Y.encode_state_as_update(d2, state_vector)
                Y.apply_update(d1, diff)

File: tests-wasm/testHelper.js
import * as Y from 'ywasm'

/**
 * @this {YDoc}
 */
Y.YDoc.prototype.transact = function(callback, origin) {
  let txn = this.writeTransaction(origin)
  try {
      return callback(txn)
  } finally {
      txn.commit()
      txn.free()
  }
};

/**
 * @param {Array<YDoc>} docs
 */
export const exchangeUpdates = docs => {
    for(let d1 of docs) {
        for(let d2 of docs) {
            if (d1 !== d2) {
                let stateVector = Y.encodeStateVector(d1)
                let diff = Y.encodeStateAsUpdate(d2, stateVector)

                Y.applyUpdate(d1, diff, "exchange")
            }
        }
    }
}